### DISCUSSION TITLE
How to use DataProvider to sync charts with different yScale domains?

### DISCUSSION THREAD

**Discussion by MarkLyck** [airbnb/visx#1646] [Q&A] [answered]
I have some charts that I would like the user to be able to compare.

In the previous chart system I used a Dual Axis chart for this, but looking into it more, it seems like it's not recommended in visx and a great comment instead says to use DataProvider and EventEmitter to synchronize multiple charts together.

The docs provide an example here: https://codesandbox.io/s/linked-tooltips-7s0jz?file=/Example.tsx which by the way is broken and it needs to install `@react-spring/web` to work.

And I implemented that method in my charts without much issue:

<img width="442" alt="Screenshot 2023-02-02 at 12 48 54" src="https://user-images.githubusercontent.com/6841110/216316899-7f5174ac-395d-4474-82ad-01df76b68e17.png">

However, the scale is completely wacky. Each chart normally has it's own y scale domain where I use `yMin -5 ` and `yMax +5 ` which produces charts that look like this:

<img width="641" alt="Screenshot 2023-01-31 at 17 03 45" src="https://user-images.githubusercontent.com/6841110/216317567-0d11bdc8-f514-4e5e-9174-bc7c14c64139.png">

Rather than the first screenshot where all the charts are flat because they naturally have different domains. But when using DataProvider it seems to override my individual yScale options on each chart?

For each chart I have a XYChart config like this:

```tsx
<XYChart
      margin={{ top: 16, right: 0, bottom: 16, left: 32 }}
      width={undefined}
      height={height}
      xScale={{ type: 'time', paddingInner: 0.5 }}
      yScale={{ type: 'linear', domain: [yMinDomain, yMaxDomain], zero: false }}
      theme={customTheme}
    >
```

Which is being completely ignored after adding the DataProvider

Is it possible e.g. to set a yScale based on the `dataKey`?

**Comment by williaster** [comment, marked as answer]
Hey @MarkLyck thanks for checking out `visx`! I'm excited someone is using these advanced features, we use them internally and they can be quite powerful. (Thanks also for noting the linked tooltip example is broken, I just updated it 🙏 )

Here's an updated version of it where each chart has independent data – this probably should be the example we use in the docs actually. https://codesandbox.io/s/linked-tooltips-independent-data-vvesr0?file=/Example.tsx

<img width="500" alt="image" src="https://user-images.githubusercontent.com/4496521/216493040-64804108-a3e2-42ec-9702-673d25d19169.png">

The key difference is that the charts only have a shared `EventEmitterProvider` – **not** a shared `DataProvider`. The `DataProvider` handles collecting all the data from child series and computing scales, which means that if charts share this they will have a single scale across all data in all charts (unless the scale config is overwritten with an explicit `domain` specifying the limits, but it would still be shared across charts).

One note about using this with different x-axis values: the event emitter emits events at a given x/y position, so if your charts have different widths it may not work and if you have different x-axis values it may look something like this (note the chart with few points doesn't update as frequently as the others due to a given point being the "closest" point for a longer time as you move the cursor)

https://codesandbox.io/s/linked-tooltips-independent-data-forked-nctceq?file=/Example.tsx (messy code/didn't clean it up)
<img width="500" alt="image" src="https://user-images.githubusercontent.com/4496521/216494856-49c0322c-fe60-4585-8cff-bc3c4a1f3191.gif">

**Reply by MarkLyck** [reply]
@williaster This is fantastic, thank you so much for the prompt help and updating the exampels! 🚀

Really loving visx XYChart so far.
